src/lib/utils.ts

```diff
@@ -3,55 +3,63 @@
 
 import Log from "./logger";
 
-/** Test if a string is ascii or not */
-export const isAscii = /^[\s\x20-\x7E]*$/;
+type Base64ConvertibleValue = string | Buffer;
 
-/**
- * Encode all ascii values in a map to base64
- * @param obj The object to encode
- * @param skip A list of keys to skip encoding
- */
-export function convertToBase64Map(obj: { data?: Record<string, string> }, skip: string[]) {
-  obj.data = obj.data ?? {};
-  for (const key in obj.data) {
-    const value = obj.data[key];
-    // Only encode ascii values
-    obj.data[key] = skip.includes(key) ? value : base64Encode(value);
+interface Base64Map {
+  data?: Record<string, Base64ConvertibleValue>;
+}
+
+/** Base64 encode all values in a map */
+export function convertToBase64Map(obj: Base64Map): Base64Map {
+  const newObj: Base64Map = { ...obj };
+  newObj.data = newObj.data ?? {};
+
+  for (const key in newObj.data) {
+    const value = newObj.data[key];
+    newObj.data[key] = base64Encode(value);
+  }
+
+  return newObj;
+}
+
+/** Base64 decode all values in a map */
+export function convertFromBase64Map(obj: Base64Map): Base64Map {
+  const newObj: Base64Map = { ...obj };
+  newObj.data = newObj.data ?? {};
+
+  for (const key in newObj.data) {
+    const value = newObj.data[key];
+    newObj.data[key] = base64Decode(value);
   }
+
+  return newObj;
 }
 
 /**
- * Decode all ascii values in a map from base64 to utf-8
- * @param obj The object to decode
- * @returns A list of keys that were skipped
+ * Decode a base64 string to a Buffer.
+ * A Buffer is used to handle binary data accurately without unintended character encoding conversions.
+ * @param data The base64 encoded string or Buffer containing binary data
+ * @returns A Buffer containing the decoded data, or an empty Buffer in case of errors
  */
-export function convertFromBase64Map(obj: { data?: Record<string, string> }) {
-  const skip: string[] = [];
-
-  obj.data = obj.data ?? {};
-  for (const key in obj.data) {
-    if (obj.data[key] == undefined) {
-      obj.data[key] = "";
-    } else {
-      const decoded = base64Decode(obj.data[key]);
-      if (isAscii.test(decoded)) {
-        // Only decode ascii values
-        obj.data[key] = decoded;
-      } else {
-        skip.push(key);
-      }
+export function base64Decode(data: Base64ConvertibleValue): Buffer {
+  try {
+    if (typeof data === "string") {
+      return Buffer.from(data, "base64");
+    } else if (Buffer.isBuffer(data)) {
+      return data;
     }
+  } catch (error) {
+    Log.error(`Error decoding base64 data: ${error}`);
```

Comment:
can you reproduce this problem in kubernetes? I wasn't able to shove invalid base64 data into a .spec.data[]